Annotato/Annotato/Views/Document/Edit/DocumentView.swift

```diff
@@ -123,7 +212,35 @@ extension DocumentView {
         return visiblePages.contains(pageContainingAnnotation)
     }
 
+    private func selectionBoxIsInVisiblePages(selectionBox: SelectionBoxView, visiblePages: [PDFPage]) -> Bool {
+        guard let centerInDocument = pdfView.documentView?.convert(selectionBox.center, to: self) else {
+            return false
+        }
+        guard let pageContainingSelectionBox = pdfView.page(for: centerInDocument, nearest: true) else {
+            return false
+        }
+        return visiblePages.contains(pageContainingSelectionBox)
+    }
+
+    private func linkLineIsInVisiblePages(linkLine: LinkLineView, visiblePages: [PDFPage]) -> Bool {
+        guard let centerInDocument = pdfView.documentView?.convert(linkLine.center, to: self) else {
+            return false
+        }
+        guard let pageContainingLinkLine = pdfView.page(for: centerInDocument, nearest: true) else {
+            return false
+        }
+        return visiblePages.contains(pageContainingLinkLine)
+    }
+
     private func bringAnnotationToFront(annotation: AnnotationView) {
         annotation.superview?.bringSubviewToFront(annotation)
     }
+
+    private func bringSelectionBoxToFront(selectionBox: SelectionBoxView) {
+        selectionBox.superview?.bringSubviewToFront(selectionBox)
+    }
+
+    private func bringLinkLineToFront(linkLine: LinkLineView) {
+        linkLine.superview?.bringSubviewToFront(linkLine)
+    }
```

Comment:
Same for these, since there's duplication now, maybe we can have an extension. You can add it to the current `UIView` extension in the `Views > Extensions` folder

```
func bringToFrontOfSuperview() {
    self.superview?.bringSubviewToFront(self)
}
```

Reply:
Ok done